app/models/sg_permissions.rb

```diff
@@ -1,22 +1,33 @@
+# encoding: UTF-8
+
 module SgPermissions
   module ClassMethods
     # sg_parent :activity
     # sg_parent :any_sequence
-    # sg_parent :any_prompt
     def sg_parent(symbol)
-      if symbol == :any_sequence
+      if (symbol == :any_sequence)
```

Comment:
This is the start of the major change in this pull request.  Of the two causes of copy failure this was the most subtle. Baked into this method were assumptions about what methods are common to all sequences. Some of those assumptions were wrong causing NameErrors.  The rescue block made the same mistake, so we were still getting NameErrors.  This new approach finds the first appropriate association that can find a parent, without blindly invoking non-existent methods.   Also "any_prompt" wasn't actually used by any caller, so that conditional block was removed.